### PR TITLE
[Feature][doc][Connector-V2][Fake] Add FakeSource connector documentation

### PR DESCRIPTION
**Pull request by zhilinli123**
## Purpose of this pull request
subTask:https://github.com/apache/seatunnel/issues/4572

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by zhilinli123** [comment]
@hailin0   @liugddx  @Hisoka-X   PTAL

File: docs/en/connector-v2/source/FakeSource.md [hunk 1]
```diff
@@ -103,172 +107,58 @@ schema = {
...
-    fields = [1, "A", 100]
-  },
-  {
-    kind = UPDATE_AFTER
-    fields = [1, "A_1", 100]
-  },
-  {
-    kind = DELETE
-    fields = [1, "A_1", 100]
+source {
```
Comment (Hisoka-X) [inline docs/en/connector-v2/source/FakeSource.md:144]: I remember FakeSource can produce data randomly, could you also add a demo case use that feature? Please.

Comment (zhilinli123) [reply docs/en/connector-v2/source/FakeSource.md:144]: Done

**Comment by zhilinli123** [comment]
PTAL: @Hisoka-X @EricJoy2048 @liugddx

**Comment by Hisoka-X** [comment]
Since there have been many changes to the Fake connector recently, it is best for us to merge this PR to the dev branch as soon as possible. Could you update this PR to date? @zhilinli123 cc @ruanwenjun

**Comment by zhilinli123** [comment]
> Since there have been many changes to the Fake connector recently, it is best for us to merge this PR to the dev branch as soon as possible. Could you update this PR to date? @zhilinli123 cc @ruanwenjun

I made some changes. Thank you for your hard work

File: docs/en/connector-v2/source/FakeSource.md [hunk 2]
```diff
@@ -16,54 +16,53 @@ just for some test cases such as type conversion or connector new feature testin
...
-| bigint.template     | list     | no       | -                       |
-| float.fake.mode     | string   | no       | range                   |
-| float.min           | float    | no       | 0                       |
-| float.max           | float    | no       | 0x1.fffffeP+127         |
-| float.template      | list     | no       | -                       |
-| double.fake.mode    | string   | no       | range                   |
-| double.min          | double   | no       | 0                       |
-| double.max          | double   | no       | 0x1.fffffffffffffP+1023 |
-| double.template     | list     | no       | -                       |
-| table-names         | list     | no       | -                       |
```
Comment (Hisoka-X) [inline docs/en/connector-v2/source/FakeSource.md:57]: you missed this.

File: docs/en/connector-v2/source/FakeSource.md [hunk 3]
```diff
@@ -16,54 +16,53 @@ just for some test cases such as type conversion or connector new feature testin
...
-
-The schema of fake data that you want to generate
-
-#### Examples
+## Source Options
+
+|        Name         |   Type   | Required |         Default         |                                                                                      Description                                                                                      |
+|---------------------|----------|----------|-------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| schema              | config   | yes      | -                       | Define Schema information                                                                                                                                                             |
+| fields              | config   | yes      | -                       | Define Field information                                                                                                                                                              |
```
Comment (Hisoka-X) [inline docs/en/connector-v2/source/FakeSource.md]: Where is `fields`?

Comment (zhilinli123) [reply docs/en/connector-v2/source/FakeSource.md]:
```
### schema [config]

#### fields [Config]

The schema of fake data that you want to generate

#### Examples

```
here

Comment (Hisoka-X) [reply docs/en/connector-v2/source/FakeSource.md]: This is a sub config of `schema`. Not for `FakeSource` root config. Please refer https://github.com/apache/seatunnel/blob/dev/docs/en/concept/schema-feature.md

Comment (zhilinli123) [reply docs/en/connector-v2/source/FakeSource.md]: done

File: docs/en/connector-v2/source/FakeSource.md [hunk 4]
```diff
@@ -16,54 +16,53 @@ just for some test cases such as type conversion or connector new feature testin
...
+| double.fake.mode    | string   | no       | range                   | The fake mode of generating float data, support `range` and `template`, default `range`，if use configured it to `template`, user should also configured `double.template` option      |
+| double.min          | double   | no       | 0                       | The min value of double data that connector generated                                                                                                                                 |
+| double.max          | double   | no       | 0x1.fffffffffffffP+1023 | The max value of double data that connector generated                                                                                                                                 |
+| double.template     | list     | no       | -                       | The template list of double type that connector generated, if user configured it, connector will randomly select an item from the template list                                       |
 | table-names         | list     | no       | -                       |
```
Comment (Hisoka-X) [inline docs/en/connector-v2/source/FakeSource.md]: description for `table-names`?

File: docs/en/connector-v2/source/FakeSource.md [hunk 5]
```diff
@@ -16,54 +16,53 @@ just for some test cases such as type conversion or connector new feature testin
...
+| double.min          | double   | no       | 0                       | The min value of double data that connector generated                                                                                                                                 |
+| double.max          | double   | no       | 0x1.fffffffffffffP+1023 | The max value of double data that connector generated                                                                                                                                 |
+| double.template     | list     | no       | -                       | The template list of double type that connector generated, if user configured it, connector will randomly select an item from the template list                                       |
+| table-names         | list     | no       | -                       | The Supports configuration of multiple data sources for example: ["test.table1", "test.table2"]                                                                                       |
```
Comment (Hisoka-X) [inline docs/en/connector-v2/source/FakeSource.md]: Check the old one https://github.com/apache/seatunnel/pull/5255/files#diff-368d0e6bf6b91efe5c90ec4eaa163de7332b3e6e64797a0fc6b27dd3d0711326L266. You losted many important information in the new one description.

File: docs/en/connector-v2/source/FakeSource.md [hunk 6]
```diff
@@ -104,190 +103,91 @@ schema = {
...
-### double.template
-
-The template list of double type that connector generated, if user configured it, connector will randomly select an item from the template list
-
-### table-names
-
-The table list that connector generated, used to simulate multi-table scenarios.
-
-This option will override the `table` option in the `schema` option.
-For example, if you configure the `table-names` option as follows, the connector will generate data for the `test.table1` and `test.table2` tables, the `database.schema.table` will be drop.
```
Comment (zhilinli123) [inline docs/en/connector-v2/source/FakeSource.md:271]: @Hisoka-X  here?

Comment (Hisoka-X) [reply docs/en/connector-v2/source/FakeSource.md:271]: Yes, including https://github.com/apache/seatunnel/pull/5255/files/0e4e841e107b721a73409f10f32ab8b7b7301f0c#diff-368d0e6bf6b91efe5c90ec4eaa163de7332b3e6e64797a0fc6b27dd3d0711326L273-L279

Comment (zhilinli123) [reply docs/en/connector-v2/source/FakeSource.md:271]: thanks review,complete modification

**Review by Hisoka-X — APPROVED** [review]

**Review by hailin0 — APPROVED** [review]

**Pull request merged**